js/classes/entities/foreground/precipitation.js

```diff
@@ -0,0 +1,133 @@
+/**
+ * Creates falling rain or snow.
+ * 
+ * @author Caleb Krauter
+ */
+class Precipitation {
+
+    /**
+     * Based on the parameters a single raindrop or snowflake is added creating precipitation.
+     * 
+     * @param {*} dir direction that the drop should be falling.
+     * @param {*} pos initial position of the drop.
+     * @param {*} type rain or snow.
+     * @param {*} scale scale of the rain or snow will vary.
+     */
+    constructor(dir, pos, type, scale) {
+        this.dir = dir;
+        this.origin = pos;
+        // Setting this.pos to (0, 0) makes rain/snow fall naturally when it starts falling.
+        this.pos = new Vector(0, 0);
+        this.type = type;
+        this.velocity = new Vector(0, 0);
+        this.scale = scale;
+        this.imgIndex = 0;
+        // Precipitation.SPRITESHEET = Precipitation.SPRITESHEET_RAIN_DOWN;
+
+        if (type === "rain") {
+            if (dir === "down") {
+                this.imgIndex = 0;
+            } else if (dir === "left") {
+                this.imgIndex = 1;
+            } else if (dir === "right") {
+                this.imgIndex = 2;
+            }
+        } else if (type === "snow") {
+            let varient = Math.ceil(Math.random() * 4);
+            switch (varient) {
+                case 1:
+                    this.imgIndex = 3;
+                    break;
+                case 2:
+                    this.imgIndex = 4;
+                    break;
+                case 3:
+                    this.imgIndex = 5;
+                    break;
+                case 4:
+                    this.imgIndex = 6;
+                    break;
+                default:
+                    break;
```

Comment:
This can be simplified as follows,

```suggestion
            const variant = Math.ceil(Math.random() * 4);
            this.imgIndex = variant + 2;
```

Reply:
Oh wow you're definitely correct. Thanks!